crates/rolldown/src/ast_scanner/side_effect_detector/mod.rs

```diff
@@ -230,8 +230,13 @@ impl<'a> SideEffectDetector<'a> {
           }
         }
       }
+      let is_destructuring = matches!(
```

Comment:
destructuring side effects detection is so complicated, let's mark them all have side effects for now. fixing https://github.com/rolldown/rolldown/pull/1566/files#diff-f40e0b0989ba3acc13279fe2e59b5fbc4e766fbcc1dca9b25daca07540275ae2R19